src/shadowbox/docker/Dockerfile

```diff
@@ -38,8 +38,15 @@ WORKDIR /root/shadowbox
 
 RUN mkdir bin && curl -SsL https://github.com/Jigsaw-Code/outline-ss-server/releases/download/v${SS_VERSION}/outline-ss-server_${SS_VERSION}_linux_x86_64.tar.gz | tar xz -C bin outline-ss-server
 COPY third_party/prometheus/prometheus ./bin/
-COPY src/shadowbox/package.json .
+COPY package.json .
```

Comment:
Why are you using the root package.json and not the shadowbox one?

Also, the root one is supposed to only have dev dependencies, so nothing will be installed with `yarn install --prod`.

Reply:
> Why are you using the root package.json and not the shadowbox one?

We need the top-level package.json for `yarn install` to resolve the workspace dependency. Without it, yarn tries to fetch the dependency from the registry.

> Also, the root one is supposed to only have dev dependencies, so nothing will be installed with yarn install --prod.

You're right in that no top-level dependencies are installed, but running `yarn install` on the root directory will install all workspace dependencies.